addons/mail/static/src/discuss/call/common/ptt_extension_service.js

```diff
@@ -48,18 +56,37 @@ export const pttExtensionHookService = {
                     break;
             }
         });
-        window.postMessage({ from: "discuss", type: "ask-is-enabled" });
+
+        /**
+         * Send a message to the PTT extension.
+         *
+         * @param {"ask-is-enabled" | "subscribe" | "unsubscribe" | "is-talking"} type
+         * @param {*} value
+         */
+        async function sendMessage(type, value) {
+            if (!isEnabled && type !== "ask-is-enabled") {
+                return;
+            }
+            const version = await versionPromise;
+            if (compareVersions(version, "1.0.0.1") <= 0) {
+                window.postMessage({ from: "discuss", type, value }, location.origin);
+            } else {
+                window.chrome?.runtime?.sendMessage(EXT_ID, { type, value });
+            }
```

Comment:
Earlier return is easier to read. In general, I'd avoid `if else` because the added indent level reduces readability quite significantly.

Here the indent level comes from keeping support of earlier versions. We could put the indent level cost on just the earlier versions, so that the current version never adds an indent level, keeping code readable for the most recent version.

```js
if (version <= 1.0.0.1) {
   window.postMessage();
   return;
}
window.chrome?.runtime?.sendMessage();
```